src/plugins/intel_cpu/src/nodes/reduce.cpp

```diff
@@ -1820,8 +1820,9 @@ void Reduce::initSupportedPrimitiveDescriptors() {
     }
 
     support_split = algorithm != Algorithm::ReduceL2 && algorithm != Algorithm::ReduceLogSumExp &&
-                    algorithm != Algorithm::ReduceSumSquare;
-    precision_change = input_prec == Precision::BF16 && output_prec == Precision::FP32;
+                    algorithm != Algorithm::ReduceSumSquare &&
+                    (input_prec == output_prec || (input_prec == Precision::BF16 && output_prec == Precision::FP32));
+    precision_change = input_prec != output_prec;
 
```

Comment:
Would be more logical to have the following:
```cpp
   precision_change = input_prec != output_prec;
   support_split = algorithm != Algorithm::ReduceL2 && algorithm != Algorithm::ReduceLogSumExp &&
                    algorithm != Algorithm::ReduceSumSquare &&
                    (!precision_change  || (input_prec == Precision::BF16 && output_prec == Precision::FP32));
```

Reply:
Good point! Applied. Thanks Maksim!